src/components/FlowRunGraphArtifactDrawer.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <p-drawer
+    class="flow-run-graph-artifact-drawer"
+    :open="artifactId !== null"
+    placement="right"
+    @update:open="close"
+  >
+    <template v-if="artifactId && artifact">
```

Comment:
nit: be a little simpler to just check for `artifact` and then use `artifact.id` rather than making this block dependent on two variables.